common/pkg/testutils/capitest/patches.go

```diff
@@ -29,6 +29,7 @@ type PatchTestDef struct {
 	ExpectedPatchMatchers []JSONPatchMatcher
 	// UnexpectedPatchMatchers used to test patches that should not be present
 	UnexpectedPatchMatchers []JSONPatchMatcher
+	AdditionalValidations   func()
```

Comment:
Added this in `PatchTestDef` as a generic function to validate anything else for these handler tests, but let me know if we don't like this.